dali/pipeline/data/tensor_list.cc

```diff
@@ -463,6 +463,8 @@ const TensorListShape<> &TensorList<Backend>::shape() const & {
 
 template <typename Backend>
 void TensorList<Backend>::set_order(AccessOrder order, bool synchronize) {
+  if (!order)
```

Comment:
So you cannot nullify the order of the tensor list?

Reply:
And.. what do we need `AccessOrder::null_stream()` for?

Reply:
I added a restriction that disallows setting the empty order to TensorList and Buffer, and allowed it to mean no-op in the APIs where it was used that way (so it can be an optional argument). For the null_stream, I think it is invalid, and if used I guess we will get an error?